drivers/vulkan/vulkan_context.cpp

```diff
@@ -937,15 +928,24 @@ Error VulkanContext::_create_instance() {
 		enabled_extension_names[enabled_extension_count++] = extension_name.ptr();
 	}
 
+	// Our application_api_version is the Vulkan version Godot is designed for
+	// We default this to our instance version but cap it on our current Vulkan header version.
+	application_api_version = instance_api_version;
```

Comment:
So this is the new logic around checking the application version, in theory we could just set this to `VK_HEADER_VERSION_COMPLETE` and strip off the patch number.

Reply:
Apparently the Vulkan spec doesn't want us to use `VK_HEADER_VERSION_COMPLETE` as `VkApplicationInfo::apiVersion`
> VK_HEADER_VERSION_COMPLETE is the complete version number of the vulkan_core.h header, comprising the major, minor, and patch versions. The major/minor values are kept synchronized with the complete version of the released Specification. This value is intended for use by automated tools to identify exactly which version of the header was used during their generation.
> 
> Applications should not use this value as their [VkApplicationInfo](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkApplicationInfo.html)::apiVersion. Instead applications should explicitly select a specific fixed major/minor API version using, for example, one of the VK_API_VERSION_*_* values.

[VK_HEADER_VERSION_COMPLETE(3)](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_HEADER_VERSION_COMPLETE.html#_c_specification)

Reply:
Application API version should be fixed indeed. We could just make it a constant/macro defined as either
a) `VK_API_VERSION_1_3`, and remember to upgrade it when we target a more recent Vulkan spec, or
b) `VK_HEADER_VERSION_COMPLETE - VK_HEADER_VERSION`, to get automatic upgrades, so to speak (I believe such an usage of `VK_HEADER_VERSION_COMPLETE` would be fine), or
c) mix both so we use `VK_API_VERSION_1_3` but add a compile-time check that compares it with the expression suggested in b) to get a reminder that we have to raise our application version constant/macro because of upgraded Vulkan headers.

Reply:
Now, regarding `instance_api_version`, it seems it's not checked anywhere. Unless there's some call whose availability depends in the instance version being > 1.1, I think we can just remove anything related to the instance version. Thoughts?

Reply:
> a) `VK_API_VERSION_1_3`, and remember to upgrade it when we target a more recent Vulkan spec, or

I think we should go with this option but use the lowest possible version for `VkApplicationInfo::apiVersion`. As far as I'm aware we don't use any core Vulkan 1.3 features, so the `VkApplicationInfo::apiVersion` should be set to `VK_API_VERSION_1_2`.

Why the lowest possible version? 
A Vulkan loader version < 1.3.228 refuses to load implicit layers that use an API version lower than `VkApplicationInfo::apiVersion`. So, if a user has the `VK_LAYER_OBS_HOOK` implicit layer with an API version of 1.2.x installed and Godot's API version is 1.3, a Vulkan loader version < 1.3.228  would refuse to load the OBS implicit layer even though Godot doesn't use any Vulkan 1.3 features. https://github.com/KhronosGroup/Vulkan-Loader/blob/master/docs/LoaderLayerInterface.md#versioning-and-activation-interactions

Reply:
Agreed. I don't know why I thought we're targeting Vulkan 1.3. 1.2 is the right version indeed.

Reply:
I'm not sure if there is any reason for us to consider Vulkan 1.3 over 1.2 though my gut feeling says there isn't a whole lot of reason not to stay on the latest 1.x version as long as we keep checking for support of availability of extensions and features. Though Sakrel has a good point if layers refuse to load though I wonder what Vulkans thinking behind this is.
For safety I will change it to 1.2 but I think this is worth further discussion.

Also agree that the more I learn about the instance version, the more I wonder if there is any reason for us to query this.